src/sys/unix/waker/eventfd.rs

```diff
@@ -42,6 +42,13 @@ impl Waker {
 
     #[allow(clippy::unused_io_amount)] // Don't care about partial writes.
     pub(crate) fn wake(&self) -> io::Result<()> {
+        // The epoll emulation on some illumos systems currently requires
+        // the eventfd to be read before an edge-triggered read event is
+        // generated.
+        // See https://www.illumos.org/issues/16700.
+        #[cfg(target_os = "illumos")]
+        self.reset();
```

Comment:
Since `self.reset()` returns an error, I think this should have a `?` at the end.